HexCtrl/src/CHexCtrl.cpp

```diff
@@ -3438,6 +3439,22 @@ void CHexCtrl::OperData(T* pData, EHexOperMode eMode, T tDataOper, ULONGLONG ull
 	auto nChunks = ullSizeData / sizeof(T);
 	for (const auto pDataEnd = pData + nChunks; pData < pDataEnd; ++pData)
 	{
+		if (fBigEndian)
+		{
+			switch (sizeof(T))
+			{
+				case 2:	
+					*pData = _byteswap_ushort(*pData);
+					break;
+				case 4:
+					*pData = _byteswap_ulong(*pData);
+					break;
+				case 8:
+					*pData = _byteswap_uint64(*pData);
+					break;
+			}
+		}

```

Comment:
In that kind of template contexts the `if constexpr` must be used, to totally annihilate runtime overhead.
Like in [this snippet](https://github.com/jovibor/HexCtrl/blob/c214ffef761aa172b215534669341bc944bae704/HexCtrl/src/Dialogs/CHexDlgDataInterp.cpp#L464).